src/app/data-planes/data/index.ts

```diff
@@ -0,0 +1,200 @@
+import { DataplaneNetworking } from './networking'
+import { Version } from './version'
+import type {
+  PaginatedApiListResponse as CollectionResponse,
+} from '@/types/api.d'
+import type {
+  DataPlaneOverview as OriginalDataplaneOverview,
+  DataPlane as OriginalDataplane,
+  DataPlaneInsight as OriginalDataplaneInsight,
+  DiscoverySubscription,
+  KumaDpVersion,
+  EnvoyVersion,
+} from '@/types/index.d'
+import {
+  compatibilityKind,
+  COMPATIBLE,
+} from '@/utilities/dataplane'
+
+export type Dataplane = OriginalDataplane & {
+  networking: DataplaneNetworking
+  tags: { label: string, value: string }[]
+  service: string
+  status: string
+  protocol: string
+  zone: string
+}
+export type DataplaneInsight = OriginalDataplaneInsight & {
+  subscriptions: DiscoverySubscription[];
+}
+
+export type DataplaneOverview = OriginalDataplaneOverview & {
+  status: string
+  // dependencies: Record<string, string>
+  lastUpdated: Date
+  version: string
+  compatible: boolean,
+  dataplane: Dataplane
+  dataplaneInsight: DataplaneInsight
+}
+
+type Label = {
+  label: string
+  value: string
+}
+type BuildVersion = KumaDpVersion | EnvoyVersion | { version: string}
+///
+
+export const Dataplane = {
+  fromJSONCollection(collection: CollectionResponse<unknown>): CollectionResponse<Dataplane> {
+    collection.items = collection.items.map((item: unknown) => {
+      return Dataplane.fromJSON(item)
+    })
+    return collection as CollectionResponse<Dataplane>
+  },
+  fromJSON(object: unknown): Dataplane {
+    const item = {
+      ...(object as OriginalDataplane),
```

Comment:
In regard to the above conversation `GeneratedDataplane.fromJSON()` would go somewhere in here.